tests/bson/bson-decimal128interface-002.phpt

```diff
@@ -0,0 +1,97 @@
+--TEST--
+MongoDB\BSON\Decimal128Interface example with MongoDB\BSON\TypeWrapper
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+class MyDecimal128 implements MongoDB\BSON\TypeWrapper, MongoDB\BSON\Decimal128Interface
+{
+    private $decimal;
+
+    public function __construct($value)
+    {
+        $this->decimal = new MongoDB\BSON\Decimal128($value);
+    }
+
+    public static function createFromBSONType(MongoDB\BSON\Type $type)
+    {
+        if ( ! $type instanceof MongoDB\BSON\Decimal128) {
+            throw new InvalidArgumentException('Cannot create MyDecimal128 from ' . get_class($type));
+        }
+
+        return new self($type);
+    }
+
+    public function toBSONType()
+    {
+        return $this->decimal;
+    }
+
+    public function __toString()
+    {
+        return (string) $this->decimal;
+    }
+}
+
+$decimal = new MongoDB\BSON\Decimal128('1234.5678');
+$myDecimal = new MyDecimal128('1234.5678');
+
+echo "\nTesting MyDecimal128 and Decimal128 are equivalent\n";
+var_dump((string) $decimal === (string) $myDecimal);
```

Comment:
Note: `__toString()` is the only way to obtain the value of `Decimal128` and `ObjectID` objects. The method is merely a convenience for Binary, Javascript, Regex, Timestamp, and UTCDateTime (note: it is the easiest way to obtain a UTCDateTime's milliseconds). It does not apply at all for MaxKey and MinKey.

I originally decided not to add it to the type-specific interfaces, as I didn't like the idea of incorporating a magic method into an interface (one could say the same about `__debugInfo()` if [PHPC-470](https://jira.mongodb.org/browse/PHPC-470) was already merged). However, I'm not wondering if it makes sense to add it. I also purposefully omitted JsonSerializable and Serializable, am plan to simply suggest that users also implement those in their custom wrapper classes if desired.